### PR TITLE
Task history

### PR DESCRIPTION
**Pull request by groupsky**
Use d3 based gantt chart adapted from https://github.com/smiller393/GanttTimeline.

Extracted controls in the page.

Still need to properly handle ajax loading errors and display loader.

Screenshot from 2020-05-19 12-22-24

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -18,7 +17,20 @@ module.exports = {
         exclude: /node_modules\//,
         loader: 'babel-loader',
         options: {
-          presets: ['@babel/preset-env', '@babel/preset-react']
+          presets: [
+            [
+              '@babel/preset-env',
+              {
+                bugfixes: true,
+                modules: false,
```
Comment (groupsky) [inline webpack.config.js]: needed for tree-shaking

File: webpack.config.js [hunk 2]
```diff
@@ -18,7 +17,20 @@ module.exports = {
         exclude: /node_modules\//,
         loader: 'babel-loader',
         options: {
-          presets: ['@babel/preset-env', '@babel/preset-react']
+          presets: [
+            [
+              '@babel/preset-env',
+              {
+                bugfixes: true,
```
Comment (groupsky) [inline webpack.config.js]: [future-proof](https://babeljs.io/docs/en/babel-preset-env#bugfixes)

File: webpack.config.js [hunk 3]
```diff
@@ -18,7 +17,20 @@ module.exports = {
...
+                modules: false,
+                useBuiltIns: false,
+                targets: {
+                  esmodules: true
```
Comment (groupsky) [inline webpack.config.js]: needed for async

File: webpack.config.js [hunk 4]
```diff
@@ -18,7 +17,20 @@ module.exports = {
...
         loader: 'babel-loader',
         options: {
-          presets: ['@babel/preset-env', '@babel/preset-react']
+          presets: [
+            [
+              '@babel/preset-env',
+              {
+                bugfixes: true,
+                modules: false,
+                useBuiltIns: false,
```
Comment (groupsky) [inline webpack.config.js]: no polyfills for now

File: rt_dashboard/javascript/src/components/history/HistoryPage.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    this._controller = new AbortController()
+
+    this.shadowRoot.getElementById('zoom-in').addEventListener('click', () => this._chart.zoomIn())
+    this.shadowRoot.getElementById('zoom-out').addEventListener('click', () => this._chart.zoomOut())
+    this.shadowRoot.getElementById('pan-left').addEventListener('click', () => this._chart.panLeft())
+    this.shadowRoot.getElementById('pan-right').addEventListener('click', () => this._chart.panRight())
+  }
+
+  async connectedCallback () {
+    const response = await getHistory({signal: this._controller.signal})
```
Comment (herra) [inline rt_dashboard/javascript/src/components/history/HistoryPage.js]: Parameter is not used

Comment (groupsky) [reply rt_dashboard/javascript/src/components/history/HistoryPage.js]: Fixed

**Review by herra — APPROVED** [review]

**Pull request merged**